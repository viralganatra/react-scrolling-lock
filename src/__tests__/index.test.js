import React from 'react';
import ScrollingLock from '../index';

const Item = (props) => <div {...props} />;

describe('ScrollingLock', () => {
    it('should render correctly', () => {
        const ItemDefault = ScrollingLock()(Item);
        const ItemClassName = ScrollingLock({
            className: 'test',
        })(Item);

        const wrapperDefault = shallow(<ItemDefault />);
        const wrapperClassName = shallow(<ItemClassName foo="bar" />);

        expect(wrapperDefault).toMatchSnapshot('default');
        expect(wrapperClassName).toMatchSnapshot('class name');
        expect(() => ScrollingLock()()).toThrowErrorMatchingSnapshot();
    });
});
