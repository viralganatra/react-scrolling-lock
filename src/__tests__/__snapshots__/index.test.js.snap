// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingLock should allow an empty class name 1`] = `
<div
  className=""
  onWheel={[Function]}
>
  <Item />
</div>
`;

exports[`ScrollingLock should render correctly 1`] = `
"Expected \\"WrappedComponent\\" provided as the first argument to ScrollLockHOC
            to be a function. Instead, received undefined."
`;

exports[`class name 1`] = `
<div
  className="test"
  onWheel={[Function]}
>
  <Item
    foo="bar"
  />
</div>
`;

exports[`default 1`] = `
<div
  onWheel={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <Item />
</div>
`;
